.github/workflows/maven.yml

```diff
@@ -0,0 +1,45 @@
+# This workflow will build a Java project with Maven
+# For more information see: https://help.github.com/actions/language-and-framework-guides/building-and-testing-java-with-maven
+
+name: Deploy P2 Updatesite
+
+on:
+  push:
+    branches: [ master ]
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up JDK 1.11
+      uses: actions/setup-java@v1
+      with:
+        java-version: 1.11
+    - name: Build with Maven
+      run: mvn -B package --file pom.xml
+    - name: Replace Updatesite
+      run: |
+        rm -r updatesite
+        git config --local user.email "tellastory73@gmail.com"
```

Comment:
ah, damn - I think, we need another user for your repo ;) - sorry, my bad

Reply:
No worries :) I imagine the user would have to be my user? (or any committer on this repo?)

Reply:
You could use this: https://github.com/PyvesB/eclipse-solargraph/blob/086fee601d1b71da94687cad09444e5d44cf0010/.github/workflows/publish.yml#L24

Reply:
> You could use this: https://github.com/PyvesB/eclipse-solargraph/blob/086fee601d1b71da94687cad09444e5d44cf0010/.github/workflows/publish.yml#L24

Thank you @PyvesB, I'll check that.

Reply:
@PyvesB Even though I meanwhile suggested another approach - I would still like to understand how your solution works.
What I understood was:
* You're using the jq tool to read info from a JSON - in this case from the payload
* That way you read the user name of the pusher (of the event that triggered the job)
* ... and the email (of the same)
* ... and set both as git credentials

Am I right so far?

The thing is I just wasn't able to get it working.
I just received return code 1 on that step.
Since I didn't have a particular error message, I activated step-debugging on my repo.
But that didn't present me w/ an error message, either.

In your repository I didn't see some import of jq.
So... what do I have to do other than simply copy/pasting your two lines into my script?

Reply:
Yes, your understanding is correct. Maybe something changed with GitHub actions in the meantime, I will have to dig around some more and double check.

Reply:
@PyvesB Thank you! 😊 If the workflow on your repo should still work, this would mean that the problem is on my side...
I‘d appreciate any update. Here or via dm on twitter.

Reply:
`GITHUB_PAYLOAD` no longer seems to be a thing. However you can now get the user name from the context: `${{ github.actor }}`. See this [documentation](https://docs.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions) for more information.

Reply:
@PyvesB Thank you for the update! So, we could use the actual actor instead of (as it is for now) hardcoded ``github-action@users.noreply.github.com`` / ``Github Action``.
We could change this for the workflow.

Even though - I (from my personal perspective) like the fact that the commit made by a Github action (and not by a person) is marked as such.

Reply:
The problem is that the [GitHub Action](https://github.com/GitHub-Action) organisation that is currently being pointed to has no actual link to GitHub actions. :)

It's quite confusing to see the Docker logo appear in my GitHub homepage for an Eclipse plugin update:
![github-action](https://user-images.githubusercontent.com/10694593/87854243-e103be00-c910-11ea-991a-5fbd5b18b605.png)

Reply:
Ah ok - now I understand your point. Yes, that's true - the "Github Action" as organization should not be "referenced". This was not intended, anyway.
I could imagine that the Docker logo is used for any user that is not recognized as a Github user for a commit that is made out of a Docker container (but that's just a guess because the Github Actions organization has a different [logo](https://github.com/actions)).

So, @AObuchow what do you think? Do you want the workflow to be changed to have the actual user listed for the updatesite commit?

Reply:
@PyvesB, @AObuchow ``github.actor``is surely better than "Github Action", but it just resolves to the user name. Also, I don't know (yet) how to get to the mail address.

<img width="275" alt="Screenshot 2020-07-19 at 11 27 48" src="https://user-images.githubusercontent.com/2838592/87871712-0cd78000-c9b3-11ea-8fde-5ec08457f4d7.png">

We could hardcode the repo's owner's name and mail address, but that would also apply to all forks of the repo...

Reply:
Would appending "@users.noreply.github.com" to the username work? Doesn't GitHub map that back to the user?

Reply:
> Would appending "@users.noreply.github.com" to the username work? Doesn't GitHub map that back to the user?

Hm, good point. I'll try that. Thanks for the hint, @PyvesB.

Reply:
Works like a charm :) - I'll create a PR for this.